staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go

```diff
@@ -138,7 +138,7 @@ func UnsafeGuessKindToResource(kind schema.GroupVersionKind) ( /*plural*/ schema
 	}
 
 	switch string(singularName[len(singularName)-1]) {
-	case "s":
+	case "s","h":
```

Comment:
I don't think we should make changes to this method at this point. This change will break any existing successful use.

Instead, we should find callers to this method and give them reasonable ways to specify the actual plural resource name to use, instead of guessing/deriving.

Reply:
I finally managed to find how to add objects with the correct gvr in the fake clientset's ObjectTracker.
The solution was to not pass those objects as parameter when calling the NewSimpleClientset function, but creating them later via the Tracker.
Anyway, this is not easy and I agree that it would be better to be able to use a more elegant solution.
The problem here lies in the fact that the NewSimpleClientset function calls the Add function of the Tracker, instead of the Create function, and will therefore end up calling this UnsafeGuessKindToResource function.
As long as we don't have a better solution, I think any improvement to UnsafeGuessKindToResource could be useful to some people, especially for testing purposes.